### PR TITLE
Allow Uploading Strategy report

### PR DESCRIPTION
**Pull request by apollo1291**
Uploads Strategy report to Secure Google Bucket

### CODE REVIEW

File: backend/siarnaq/api/teams/models.py [hunk 1]
```diff
@@ -198,6 +198,9 @@ class TeamProfile(models.Model):
     avatar_uuid = models.UUIDField(default=uuid.uuid4)
     """A unique ID to identify each new avatar upload."""
 
+    """has_report = models.BooleanField(default=False)"""
```
Comment (apollo1291) [inline backend/siarnaq/api/teams/models.py:201]: Didn't want to mess with migrations. but could be useful

File: backend/siarnaq/api/teams/views.py [hunk 2]
```diff
@@ -173,3 +176,28 @@ def avatar(self, request, pk=None, *, episode_id):
...
+        detail=False,
+        methods=["post"],
+        serializer_class=TeamReportSerializer,
+        permission_classes=(IsAuthenticated, IsEpisodeAvailable),
+    )
+    def report(self, request, pk=None, *, episode_id):
+        """Update team strategy report"""
+        team = get_object_or_404(self.get_queryset(), members=request.user)
+        profile = team.profile
+        print(request.data)
```
Comment (apollo1291) [inline backend/siarnaq/api/teams/views.py:191]: will remove

**Comment by n8kim1** [comment]
(ellington tells me this is ready for review, should still be draft tho cuz small debug prints, but that's all. go to town!
@ellington sorry if i stepped on your toes)

File: frontend/src/api.js [hunk 3]
```diff
@@ -320,9 +320,9 @@ class Api {
       });
   }
 
-  static avatarUpload(avatar_file, callback) {
+  static avatarUpload(report_file, callback) {
```
Comment (j-mao) [inline frontend/src/api.js:323]: Too much find-replace? Also for `teamAvatarUpload` below.

File: frontend/src/views/account.js [hunk 4]
```diff
@@ -68,6 +68,7 @@ class Account extends Component {
       //   loaded: 0,
       // });
     }
+    // account for report upload as well
```
Comment (j-mao) [inline frontend/src/views/account.js:71]: Did you mean to put this on the team page?

File: backend/siarnaq/api/teams/views.py [hunk 5]
```diff
@@ -173,3 +176,28 @@ def avatar(self, request, pk=None, *, episode_id):
...
         return Response(None, status=status.HTTP_204_NO_CONTENT)
+
+    @extend_schema(responses={status.HTTP_204_NO_CONTENT: None})
+    @action(
+        detail=False,
+        methods=["post"],
+        serializer_class=TeamReportSerializer,
+        permission_classes=(IsAuthenticated, IsEpisodeAvailable),
+    )
+    def report(self, request, pk=None, *, episode_id):
```
Comment (j-mao) [inline backend/siarnaq/api/teams/views.py:187]: Not super important, but I might suggest copying over some of the structure of `UserViewSet.resume`, which would let them download their report too! Would essentially consist of changing `methods` to `["get", "put"]` and calling `titan.get_object` in the `get` case.

**Comment by n8kim1** [comment]
@apollo1291 go run the linter! I don't remember the exact command but it's linked in our top-level readme somewhere

**Review by n8kim1 — COMMENTED** [review]
at the very least, could we show a "report uploaded!" similar to what we do for resume?

**Review by n8kim1 — APPROVED** [review]

**Pull request merged**